florist/tests/unit/app/jobs/page.test.tsx

```diff
@@ -0,0 +1,183 @@
+) {
+    const data = {
+        model: model,
+        server_address: server_address,
+        clients_info: client_services_addresses.map(
+            (client_services_address) => ({
+                service_address: client_services_address,
+            }),
+        ),
+    };
+    return data;
+}
+
+function setupMock(
+    valid_statuses: Array<string>,
+    data: Array<object>,
+    error: boolean,
+    isLoading: boolean,
+) {
+    useJobStatus.mockImplementation((status: string) => {
+        if (valid_statuses.includes(status)) {
+            return {
+                data,
+                error,
+                isLoading,
+            };
+        } else {
+            return {
+                data: [],
+                error: false,
+                isLoading: false,
+            };
+        }
+    });
+}
+
+describe("List Jobs Page", () => {
+    it("Renders Page Title correct", () => {
+        setupMock([], [], false, false);
+        const { container } = render(<Page />);
+        const h1 = container.querySelector("h1");
+        expect(h1).toBeInTheDocument();
+        expect(h1).toHaveTextContent("Job Status");
+    });
+
+    it("Renders Status Components Headers", () => {
+        const data = [
+            mock_job_data("MNIST", "localhost:8080", ["localhost:7080"]),
+        ];
+        setupMock(
+            [
+                "NOT_STARTED",
+                "IN_PROGRESS",
+                "FINISHED_SUCCESSFULLY",
+                "FINISHED_WITH_ERROR",
+            ],
+            data,
+            false,
+            false,
+        );
+        const { getByTestId } = render(<Page />);
+        const ns_status = getByTestId("status-header-NOT_STARTED");
+        const ip_status = getByTestId("status-header-IN_PROGRESS");
+        const fs_status = getByTestId("status-header-FINISHED_SUCCESSFULLY");
+        const fe_status = getByTestId("status-header-FINISHED_WITH_ERROR");
+        expect(ns_status).toBeInTheDocument();
+        expect(ns_status).toHaveTextContent("Not Started");
+        expect(ip_status).toBeInTheDocument();
+        expect(ip_status).toHaveTextContent("In Progress");
+        expect(fs_status).toBeInTheDocument();
+        expect(fs_status).toHaveTextContent("Finished Successfully");
+        expect(fe_status).toBeInTheDocument();
+        expect(fe_status).toHaveTextContent("Finished with Error");
+    });
+
+    it("Renders Status Table With Table with Data", () => {
+        const data = [
+            mock_job_data("MNIST", "localhost:8080", ["localhost:7080"]),
+        ];
+        setupMock(
+            [
+                "NOT_STARTED",
+                "IN_PROGRESS",
+                "FINISHED_SUCCESSFULLY",
+                "FINISHED_WITH_ERROR",
+            ],
+            data,
+            false,
+            false,
+        );
+        const { getByTestId } = render(<Page />);
+        const ns_table = getByTestId("status-table-NOT_STARTED");
```

Comment:
You can make the code below into a loop to avoid code repetition:
```javascript
        const testIds = [
            "status-table-NOT_STARTED",
            "status-table-IN_PROGRESS",
            "status-table-FINISHED_SUCCESSFULLY",
            "status-table-FINISHED_WITH_ERROR",
        ]

        for(let i in testIds) {
            const jobsTable = getByTestId(testIds[i]);

            expect(getByText(jobsTable, "Model")).toBeInTheDocument();
            expect(getByText(jobsTable, "Server Address")).toBeInTheDocument();
            expect(
                getByText(jobsTable, "Client Service Addresses"),
            ).toBeInTheDocument();
            expect(getByText(jobsTable, "MNIST")).toBeInTheDocument();
            expect(getByText(jobsTable, "localhost:8080")).toBeInTheDocument();
            expect(getByText(jobsTable, "localhost:7080")).toBeInTheDocument();
        }
```

Reply:
Good idea! Not sure why I didn't do this in the first place.